src/canWrapper.cc

```diff
@@ -724,6 +724,41 @@ Napi::Array getImageElements(const Napi::CallbackInfo& info) {
     return elements;
 }
 
+Napi::Object getTimestampsForAllReceivedMessages(const Napi::CallbackInfo& info) {
+    Napi::Env env = info.Env();
+    std::string descriptor = info[0].As<Napi::String>().Utf8Value();
+
+    std::shared_ptr<rev::usb::CANDevice> device;
+
+    { // This block exists to define how long we hold canDevicesMtx
+        std::scoped_lock lock{canDevicesMtx};
```

Comment:
This could be a std::lock_guard since we are only locking a single mutex, but its fine as is.

Reply:
I do like that scoped_lock makes it clearer that it unlocks when it goes out of scope, but that's just my C++-novice self talking.

Reply:
I did some more reading last night, and it seems that 'modern' usage favors scoped_lock over lock_guard. It sounds like they would have changed lock_guard if it hadn't been for backward compatibility. So, it typical C++ fashion we get another function ;)